.github/workflows/codeql-analysis.yaml

```diff
@@ -0,0 +1,67 @@
+# For most projects, this workflow file will not need changing; you simply need
+# to commit it to your repository.
+#
+# You may wish to alter this file to override the set of languages analyzed,
+# or to provide custom queries or build logic.
+#
+# ******** NOTE ********
+# We have attempted to detect the languages in your repository. Please check
+# the `language` matrix defined below to confirm you have the correct set of
+# supported CodeQL languages.
+#
+name: "CodeQL"
+
+on:
+  push:
+    branches: [ master, '**-release-**']
+  pull_request:
+    # The branches below must be a subset of the branches above
+    branches: [ master, '**-release-**' ]
```

Comment:
If we're triggering builds on push but only building the [cluster-autoscaler](https://github.com/kubernetes/autoscaler/pull/7427/files#diff-641919750b0e7d8f9693ecb8a725fef012b1f0d6337470f35fb1610b4137246fR63), we can use [including-paths](https://docs.github.com/en/actions/writing-workflows/workflow-syntax-for-github-actions#example-including-paths). This way, the workflow won't run unnecessarily if someone pushes changes to the vertical-autoscaler (for example), as it wouldn't be relevant.

Reply:
Added. Please take a look.

Reply:
Looks good :)